### ISSUE TITLE
USWDS-Site - Layout grid: Mixin guidance doesn't match other utilities

### ISSUE THREAD

**Issue by mahoneycm** [uswds/uswds-site#2210]
### Is your feature request related to a problem? Please describe.

The layout grid [mixin guidance](https://designsystem.digital.gov/utilities/layout-grid/#sass-mixins-2) does not follow patterns set by other utility classes and generally could be improved.

### Describe the solution you'd like

Convert to mixin table similar to `flex` [utility mixin guidance](https://designsystem.digital.gov/utilities/flex/#utility-mixins-2).

### Screenshots

**Layout grid**
![image](https://github.com/uswds/uswds-site/assets/25211650/67245f8a-4d50-4dab-afa2-c03bc20fcdef)

![image](https://github.com/uswds/uswds-site/assets/25211650/b5b70070-b7a6-41f8-89d4-108a5a7186fb)

**Display flex**
![image](https://github.com/uswds/uswds-site/assets/25211650/e59ab0fc-d81c-40a0-9943-5a3c0bb05a12)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [X] I checked the [current issues](https://github.com/uswds/uswds-site/issues) for duplicate feature requests.

**Comment by amyleadem** [comment]
@sarah-sch It would be great to get your feedback on if this should be changed, and if so, how. I've also tagged @mahoneycm  because he is familiar with the issue and can help explain.

**Comment by sarah-sch** [comment]
Discussed with @amyleadem and @mahoneycm on 8/7/2023. Recommend revising layout grid guidance because current code-heavy format is not accessible. Discussed with Amy and Charle that urgency of task could be elevated to around 6.

**Comment by brunerae** [comment]
@mahoneycm based on recent activity in the linked PR, I am moving this into our Sprint.
